Firestore/Source/API/FIRQuery.m

```diff
@@ -256,6 +256,82 @@ - (FIRQuery *)queryWhereFieldPath:(FIRFieldPath *)path isGreaterThanOrEqualTo:(i
                                  value:value];
 }
 
+- (FIRQuery *)queryWherePredicate:(NSPredicate *)predicate {
+  if ([predicate isKindOfClass:[NSComparisonPredicate class]]) {
+    // Comparison predicate
+    NSComparisonPredicate *comparison = (NSComparisonPredicate *)predicate;
+    if (comparison.comparisonPredicateModifier != NSDirectPredicateModifier) {
+      FSTThrowInvalidArgument(@"Invalid query. Predicate cannot have an "
+                              "aggregate modifier.");
+    }
+    NSString *path;
+    id value = nil;
+    if (comparison.leftExpression.keyPath == nil &&
+        comparison.rightExpression.keyPath == nil) {
+      FSTThrowInvalidArgument(@"Invalid query. Predicate comparison must have "
+                              "a key path on either side of its expression.");
+    } else if (comparison.leftExpression.keyPath == nil) {
+      path = comparison.rightExpression.keyPath;
+      value = comparison.leftExpression.constantValue;
+    } else {
+      path = comparison.leftExpression.keyPath;
+      value = comparison.rightExpression.constantValue;
+    }
+    if (value == nil) {
+      FSTThrowInvalidArgument(@"Invalid query. Predicate comparison must have "
+                              "a constant on its expression.");
+    }
+    switch (comparison.predicateOperatorType) {
+      case NSEqualToPredicateOperatorType:
+        return [self queryWhereField:path isEqualTo:value];
+        break;
+      case NSLessThanPredicateOperatorType:
```

Comment:
This doesn't appear to take into account the original direction of the query despite the fact that you allow the keyPath to be on either side of the expression.

These two statements aren't equivalent and yet your code code doesn't seem distinguish between them:
  * `x < 5`
  * `5 < x`

Please add a test for the reverse ordering.